tycho-core/src/main/java/org/eclipse/tycho/core/ee/StandardExecutionEnvironment.java

```diff
@@ -110,29 +110,56 @@ public class StandardExecutionEnvironment implements Comparable<StandardExecutio
         this.logger = logger;
     }
 
-    private Set<String> readFromToolchains(Toolchain toolchain) {
+    static JavaInfo readFromToolchains(Toolchain toolchain, Logger logger) {
         if (toolchain == null) {
-            return Collections.emptySet();
+            return new JavaInfo(-1, Collections.emptySet());
         }
         String java = toolchain.findTool("java");
         if (java == null) {
-            return Collections.emptySet();
+            return new JavaInfo(-1, Collections.emptySet());
         }
         Set<String> res = new HashSet<>();
+        int version = -1;
         try {
             ProcessBuilder builder = new ProcessBuilder(java, "-jar",
                     getSystemPackagesCompanionJar().getAbsolutePath());
             try (BufferedReader reader = new BufferedReader(
                     new java.io.InputStreamReader(builder.start().getInputStream(), Charset.defaultCharset()))) {
-                String line = null;
+                String line = reader.readLine();
+                try {
+                    if (line != null) {
+                        //for old vms < java 9 we might get no response at all
+                        version = Integer.parseInt(line);
+                    }
+                } catch (NumberFormatException e) {
+                    StringBuilder sb = new StringBuilder(line);
+                    while ((line = reader.readLine()) != null) {
+                        sb.append(System.lineSeparator());
+                        sb.append(line);
+                    }
+                    logger.debug("[ReadPackagesFromToolchains] Can't read java version for " + java
+                            + ", full output was: " + sb);
+                    return new JavaInfo(-1, List.of());
```

Comment:
Wouldn't it be more explicit to just return nothing here instead of returning an incorrect object?